darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -393,6 +394,20 @@ def register_figures(self):
         ## Per-Module Statistics
         ################################
         for mod in self.report.modules:
+            if mod in ["POSIX", "MPI-IO"]:
```

Comment:
Why does the access table include modules that the plots do not? For example, `ior_hdf5_example.darshan` is missing an access plot for `H5D`. Well, I can think of a reason, but I don't see it mentioned here or in the matching issue, so that seems like something obvious to clarify, and what you plan to do about it. Probably want to check with the team about this module, since the old report seems to include neither the table nor the plot for it.

![image](https://user-images.githubusercontent.com/7903078/151910401-8d1ec262-9145-40e6-8433-bf63f90f85ac.png)

Reply:
Thanks, that's a good question. And thanks for the comment @paralepsis 

I was going off the "supported_mods" listed in `experimental/aggregators/mod_agg_iohist.py`, but it looks like we can add `H5D` without much effort:

```diff
diff --git a/darshan-util/pydarshan/darshan/cli/summary.py b/darshan-util/pydarshan/darshan/cli/summary.py
index f7da02d..32c4fae 100644
--- a/darshan-util/pydarshan/darshan/cli/summary.py
+++ b/darshan-util/pydarshan/darshan/cli/summary.py
@@ -394,7 +394,7 @@ class ReportData:
         ## Per-Module Statistics
         ################################
         for mod in self.report.modules:
-            if mod in ["POSIX", "MPI-IO"]:
+            if mod in ["POSIX", "MPI-IO", "H5D"]:
                 access_hist_description = (
                     "Read/write operations grouped by access size. Most frequent "
                     "access sizes are featured in the <i>Common Access Sizes</i> table."
@@ -408,7 +408,6 @@ class ReportData:
                     fig_width=350,
                 )
                 self.figures.append(access_hist_fig)
-            if mod in ["POSIX", "MPI-IO", "H5D"]:
                 if mod == "MPI-IO":
                     com_acc_tbl_description = (
                         "NOTE: MPI-IO accesses are given in "
diff --git a/darshan-util/pydarshan/darshan/experimental/aggregators/mod_agg_iohist.py b/darshan-util/pydarshan/darshan/experimental/aggregators/mod_agg_iohist.py
index d9bb9ee..e9fcf35 100644
--- a/darshan-util/pydarshan/darshan/experimental/aggregators/mod_agg_iohist.py
+++ b/darshan-util/pydarshan/darshan/experimental/aggregators/mod_agg_iohist.py
@@ -12,7 +12,7 @@ def mod_agg_iohist(self, mod, mode='append'):
     """
 
     # sanitation and guards
-    supported = ["POSIX", "MPI-IO"]
+    supported = ["POSIX", "MPI-IO", "H5D"]
     if mod not in supported:
         raise Exception("Unsupported mod_name for aggregated iohist.")
``` 


And it seems to generate the figure just fine:


![Screen Shot 2022-02-01 at 1 05 01 PM](https://user-images.githubusercontent.com/46797896/152042849-d9092678-fc89-40ef-ba86-304f3e1b05e9.png)

For the "Common Access Sizes" table, when I was figuring out how to retrieve the module data for `POSIX` and `MPI-IO`, I found that `H5D` didn't require any extra work to support. So that's why it has some extra module coverage at the moment. 

As for a solution, based on the diff above I think we can add support for `H5D` fairly easily. We will have to increase the scope of the current tests of course. I could open a separate PR for that.